src/test/scala/inox/solvers/SemanticsSuite.scala

```diff
@@ -54,8 +54,17 @@ class SemanticsSuite extends FunSuite {
 
     check(s, BooleanLiteral(true),   BooleanLiteral(true))
     check(s, BooleanLiteral(false),  BooleanLiteral(false))
-    check(s, IntLiteral(0),          IntLiteral(0))
-    check(s, IntLiteral(42),         IntLiteral(42))
+    check(s, Int8Literal(0),         Int8Literal(0))
+    check(s, Int8Literal(58),        Int8Literal(58))
+    check(s, Int32Literal(0),        Int32Literal(0))
+    check(s, Int32Literal(42),       Int32Literal(42))
+    // FIXME currently BV 8/16/32 are supported by BitvectorEncoder.
+    // check(s, BVLiteral(0, 13),       BVLiteral(0, 13))
+    // check(s, BVLiteral(58, 64),      BVLiteral(58, 64))
+    // FIXME currently BV 8/32 are supported by Z3Native.
+    // check(s, BVLiteral(0, 16),       BVLiteral(0, 16))
+    // check(s, BVLiteral(58, 16),      BVLiteral(58, 16))
+    // NOTE When/if this is fixed, add relevant tests in the sections below.
```

Comment:
This is probably not at the top of our priority, but do note the current limitations.